Source/Microsoft.Teams.Apps.CompanyCommunicator.Prep.Func/PreparingToSend/Activities/GetRecipientsActivity.cs

```diff
@@ -35,10 +37,60 @@ public GetRecipientsActivity(SentNotificationDataRepository sentNotificationData
         /// <param name="notification">notification.</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
         [FunctionName(FunctionNames.GetRecipientsActivity)]
-        public async Task<IEnumerable<SentNotificationDataEntity>> GetRecipientsAsync([ActivityTrigger] NotificationDataEntity notification)
+        public async Task<(IEnumerable<SentNotificationDataEntity>, TableContinuationToken)> GetRecipientsAsync([ActivityTrigger] NotificationDataEntity notification)
         {
-            var recipients = await this.sentNotificationDataRepository.GetAllAsync(notification.Id);
-            return recipients;
+            if (notification == null)
+            {
+                throw new ArgumentNullException(nameof(notification));
+            }
+
+            var results = await this.sentNotificationDataRepository.GetByCountAsync(notification.Id, 1000);
+            var recipients = new List<SentNotificationDataEntity>();
+            recipients.AddRange(results.Item1);
+            while (results.Item2 != null && recipients.Count < this.maxResultSize)
+            {
+                results = await this.sentNotificationDataRepository.GetByTokenAsync(results.Item2, notification.Id);
```

Comment:
Do we need to specify the count again here? What caps the # of results in this call-- is that encoded in the continuation token?

Reply:
By default it is 1000 but capping the results makes more sense.